mavros/src/plugins/setpoint_raw.cpp

```diff
@@ -217,6 +238,47 @@ class SetpointRawPlugin : public plugin::PluginBase,
 				body_rate,
 				req->thrust);
 	}
+
+	void rpyt_cb(const mav_msgs::RollPitchYawrateThrustConstPtr msg)
+	{
+	
+		if(ignore_rpyt_messages_){
+			ROS_FATAL("Recieved roll_pitch_yaw_thrust_rate message, but ignore_rpyt_messages_ is true: "
+	 	        "the most likely cause of this is a failure to specify the thrust_scaling_factor,	"
+			"yaw_rate_scaling_factor or system_mass_kg parameters");
+		     	return;
+		}
+
+		//the masks are much more limited than the docs would suggest so we don't use them
+		uint8_t type_mask = 0;
+
+		geometry_msgs::Quaternion orientation = tf::createQuaternionMsgFromRollPitchYaw( msg->roll, msg->pitch, 0);
+		
+		double thrust = std::min(1.0, std::max(0.0, msg->thrust.z * thrust_scaling_ * system_mass_kg_));
+		Eigen::Quaterniond desired_orientation;
+		Eigen::Vector3d body_rate;
+		tf::quaternionMsgToEigen(orientation, desired_orientation);
+		
+		// Transform desired orientation to represent aircraft->NED,
+		// MAVROS operates on orientation of base_link->ENU
+		
+		auto ned_desired_orientation = ftf::transform_orientation_enu_ned(ftf::transform_orientation_baselink_aircraft(desired_orientation));
+		
+		body_rate.x() = 0;
+		body_rate.y() = 0;
+		body_rate.z() = -yaw_rate_scaling_*msg->yaw_rate;
```

Comment:
`yaw_rate_scaling * ftf::transform_frame_yaw(msg->yaw_rate)`

Reply:
clarify what happends on transform_frame_yaw ?

Reply:
https://github.com/mavlink/mavros/blob/master/mavros/include/mavros/frame_tf.h#L143

Reply:
more cleaner, and does the required job